src/components/wizards/create-vm/CreateVmWizard.js

```diff
@@ -0,0 +1,233 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Wizard } from 'patternfly-react';
+import { get } from 'lodash';
+import { createVM } from '../../../k8s/request';
+import { FormFactory } from '../../forms/FormFactory';
+import { isPositiveNumber } from '../../../utils/validation';
+
+export class CreateVmWizard extends React.Component {
+  state = {
+    activeStepIndex: 0,
+    basicVmSettings: {},
+    wizardValid: false
+  };
+
+  onFormChange = (newValue, target) => {
+    let validMsg;
+
+    if (this.basicFormFields[target].validate) {
+      validMsg = this.basicFormFields[target].validate(newValue);
+    }
+    if (this.basicFormFields[target].required && newValue.trim().length === 0) {
+      validMsg = 'is required';
```

Comment:
I wonder if we should do something with the strings to make it easier later on to i18n.

Reply:
maybe we could address that in followup patch ? :) Im not sure what solution for i18l we will want to use. It would be nice to keep aligned with openshift console but they do not have any i18l yet.

Reply:
The `openshift/console` project (frontend part) currently doesn't seem to use any i18n mechanism, there are English strings hardcoded in their components.

I'd like to avoid adding complexity until we really need it. For now, I would focus on writing simple, reusable components (and break down bigger components into smaller bits for that matter) along with associated `.test.js` and `.stories.js` files.